cumulusci/tasks/bulkdata/extract.py

```diff
@@ -142,6 +143,9 @@ def _soql_for_mapping(self, mapping):
 
     def _run_query(self, soql, mapping):
         """Execute a Bulk or REST API query job and store the results."""
+
+        soql = append_where_clause(soql=soql, object_config=mapping)
```

Comment:
Could you move this into the `_soql_for_mapping` method? That method is reponsible for building the SOQL query, and this one is responsible for executing it.

Also, could you move the `append_where_clause` function into this module, at the bottom? I consider it a feature of this task, not a plugin, so I don't think it belongs in plugins/append_where_clause.py